compiler/ast/ast.go

```diff
@@ -49,19 +49,19 @@ type Expression interface {
 // the native Go types) and value is a string representation of that value that
 // must conform to the provided type.
 type Literal struct {
-	Op    string `json:"op"`
+	Op    string `json:"op" unpack:""`
 	Type  string `json:"type"`
 	Value string `json:"value"`
 }
 
 type Search struct {
-	Op    string  `json:"op"`
+	Op    string  `json:"op" unpack:""`
 	Text  string  `json:"text"`
 	Value Literal `json:"value"`
 }
 
 type UnaryExpression struct {
-	Op       string     `json:"op"`
+	Op       string     `json:"op" unpack:"UnaryExpr"`
```

Comment:
Maybe align all the `ast` type names with their corresponding `op` values so all of these can be `unpack:""`?

Reply:
This has been on my to-do list for a while but it breaks the app so I was waiting for the right moment.

Reply:
Agreed.  I'll put this PR up after I merge sql (hopefully soon!)

Reply:
issue #2222